### PR TITLE
Tr/update docs folderstructure

### LINKED ISSUE #712: `docs/src/folderstructure.md` is outdated
I've never realized we had this page, and it is now a little outdated.

### PR DESCRIPTION
**Pull request by imreddyTeja**
## Purpose 
The `docs/src/folderstructure.md` is very outdated.
This pr updates it.
closes #712 

## Content

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: docs/src/folderstructure.md [hunk 1]
```diff
@@ -1,52 +1,124 @@
...
-[ClimaLand](https://github.com/CliMA/ClimaLand.jl) home directory has 5 main folders:
+[ClimaLand](https://github.com/CliMA/ClimaLand.jl) home directory has 6 main folders:
 
 - docs: contains files to generate [the documentation website](clima.github.io/ClimaLand.jl/).
-- experiments: contains simple runs of `ClimaLand` models. 
-- parameters: contains a file to retrieve constants such as avogadro's number, the speed of light, etc. 
-- src: contains the code of `ClimaLand` models. 
+- experiments: contains simple runs of `ClimaLand` models.
+- src: contains the code of `ClimaLand` models.
+- ext: contains a package extension for Nueral Snow and an extension with constructors for model paramaters
```
Comment (AlexisRenchon) [inline docs/src/folderstructure.md:8]: some typo here: Neural snow, parameters

**Review by AlexisRenchon — APPROVED** [review]
lgtm, thanks for doing this!

File: docs/src/folderstructure.md [hunk 2]
```diff
@@ -1,52 +1,124 @@
 # ClimaLand folder structure
 
-[ClimaLand](https://github.com/CliMA/ClimaLand.jl) home directory has 5 main folders:
+[ClimaLand](https://github.com/CliMA/ClimaLand.jl) home directory has 6 main folders:
 
 - docs: contains files to generate [the documentation website](clima.github.io/ClimaLand.jl/).
-- experiments: contains simple runs of `ClimaLand` models. 
-- parameters: contains a file to retrieve constants such as avogadro's number, the speed of light, etc. 
-- src: contains the code of `ClimaLand` models. 
+- experiments: contains simple runs of `ClimaLand` models.
```
Comment (Sbozzolo) [inline docs/src/folderstructure.md:6]:
```suggestion
- experiments: contains simple runs of `ClimaLand` models. Most of these experiments is run as part of our CI.
```

File: docs/src/folderstructure.md [hunk 3]
```diff
@@ -1,52 +1,124 @@
...
+- ext: contains a package extension for Nueral Snow and an extension with constructors for model paramaters
+- lib: contains two packages that extend ClimaLand functionality: ClimaLandSimulations and ClimaLandDashboards
 - test: contains [unit tests](https://en.wikipedia.org/wiki/Unit_testing), which are meant to ensure small pieces of ClimaLand source code work as intended before merging pull requests.
 
-and 3 GitHub actions folders. GitHub actions are `.yml` files, which are bash scripts that runs on a remote computer on each git push. 
-- .buildkite: contains a script building outputs such as figures from experiments and tests folders. These runs are carried out as part of CI and must run without error in order to merge a PR. 
-- .dev: contains useful tools for developers, such as a format checker for Julia (which is run as part of CI and must pass before a PR can be merged into main). 
-- .github: contains various scripts, for example, this documentation website is built each time a change is push to `ClimaLand`. 
+and 3 GitHub actions folders. GitHub actions are `.yml` files, which are bash scripts that runs on a remote computer on each git push.
```
Comment (Sbozzolo) [inline docs/src/folderstructure.md:12]:
```suggestion
and 3 GitHub actions folders. GitHub actions are `.yml` files that define workflows that are run at each pull request.
```

**Review by Sbozzolo — COMMENTED** [review]
Thank you!

We should try seeing how we can maintain this more updated.

File: docs/src/folderstructure.md (hunk 1 above)
Comment (imreddyTeja) [inline docs/src/folderstructure.md:8]:
```suggestion
- ext: contains a package extension for Neural Snow and an extension with constructors for model parameters
```

**Pull request merged**